src/group_theory/submonoid/inverses.lean

```diff
@@ -0,0 +1,163 @@
+/-
+Copyright (c) 2021 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+
+import group_theory.submonoid.operations
+
+/-!
+
+# Submonoid of inverses
+
+Given a submonoid `N` of a monoid `M`, we define the submonoid `N.left_inv` as the submonoid of
+left inverses of `N`. When `M` is commutative, we may define `from_comm_left_inv : N.left_inv →* N`
+since the inverses are unique. When `N ≤ is_unit.submonoid M`, this is precisely
+the pointwise inverse of `N`, and we may define `left_inv_equiv : S.left_inv ≃* S`.
+
+For the pointwise inverse of submonoids of groups, please refer to
+`group_theory.submonoid.pointwise`.
```

Comment:
Can you add a proof that when `N` is a submonoid of a group `G`, that `N.left_inv = N\inv`?